src/nvim/CMakeLists.txt

```diff
@@ -183,7 +189,7 @@ foreach(sfile ${NEOVIM_SOURCES}
     DEPENDS "${HEADER_GENERATOR}" "${sfile}"
     )
   list(APPEND NEOVIM_GENERATED_SOURCES "${gf1}")
-  list(APPEND NEOVIM_GENERATED_SOURCES "${gf2}")
```

Comment:
This code works only because `gf1` and `gf2` files are created by one command. Neovim executables and libraries *do* depend on `*.h.generated.h` files, but in your code they do not depend on ${NEOVIM_GENERATED_HEADERS}. I had two lines which appended to both NEOVIM_GENERATED_HEADERS and NEOVIM_GENERATED_SOURCES for a reason.

Though it should be cleaner if all files listed in NEOVIM_GENERATED_HEADERS are excluded from NEOVIM_GENERATED_SOURCES up until NEOVIM_GENERATED_HEADERS is appended to NEOVIM_GENERATED_SOURCES when it stopped being populated.